server/service/schedule/schedule.go

```diff
@@ -0,0 +1,219 @@
+package schedule
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/getsentry/sentry-go"
+	kitlog "github.com/go-kit/kit/log"
+	"github.com/go-kit/log/level"
+)
+
+type schedule struct {
+	ctx        context.Context
+	name       string
+	instanceID string
+	Logger     kitlog.Logger
+
+	muChecks       sync.Mutex
+	configInterval time.Duration
+	schedInterval  time.Duration
```

Comment:
`schedInterval` sets the interval on which the jobs in the schedule will be run. By contrast, `configInterval` defines how often the schedule will run a config check for changes to the schedule interval. This is meant to address use cases such as vulnerabilities where we allow changes to periodicity in the app config.